.github/workflows/towncrier-run.yml

```diff
@@ -0,0 +1,62 @@
+name: Run towncrier Check
+
+#######
+# Runs `tox -e towncrier-check` for the repository.
+#######
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: 'Python version to use; defaults to latest Python release.'
+        default: '3.X'
+        type: string
+      repository:
+        description: 'GitHub repository to checkout; defaults to repo running this workflow.'
+        default: ${{ github.repository }}
+        type: string
+      tox-source:
+        description: 'The arguments for `pip install` to install tox; use ./path/to/package[dev] for the repo''s pinned version.'
+        default: '.[dev]'
+        type: string
+      runner-os:
+        description: 'Runner OS to use to run tox; defaults to ubuntu-latest'
+        default: 'ubuntu-latest'
+        type: string
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  towncrier:
+    name: Towncrier Checks
+    runs-on: ${{ inputs.runner-os }}
+    timeout-minutes: 10
+    steps:
+
+    - name: Checkout Default Branch
+      uses: actions/checkout@v3.3.0
+      with:
+        repository: ${{ inputs.repository }}
+        ref: ${{ github.event.repository.default_branch }}
+        fetch-depth: 0
+
+    - name: Checkout Current Branch
+      uses: actions/checkout@v3.3.0
+      with:
+        repository: ${{ inputs.repository }}
+        fetch-depth: 0
+
+    - name: Setup Python
+      uses: actions/setup-python@v4.5.0
+      with:
+        python-version: ${{ inputs.python-version }}
+        cache: 'pip'
+        cache-dependency-path: '**/setup.cfg'
```

Comment:
I wonder if we might want to add `pyproject.toml` and `setup.py` to this list as well. We're still using `setup.py` in toga; I imagine that a move to full `pyproject.toml` format in the future is inevitable (either as part of a setuptools upgrade, or a move to hatch), so including those files would seem like a good protective mechanism.

Reply:
Added `**/pyproject.toml` as a path.